src/py42/clients/settings/device_settings.py

```diff
@@ -0,0 +1,415 @@
+            self._filename_exclusions.clear()
+            self._filename_exclusions.extend(value)
+        else:
+            raise AttributeError("filename exclusions must be a list/tuple.")
+
+    @property
+    def destinations(self):
+        """Returns a dict of the destinations used for backup for the backup set. Dict
+        keys are the destination guids, values are the destination names.
+        """
+        destination_dict = {}
+        for d in self.data["destinations"]:
+            guid = d["@id"]
+            dest_name = self._manager.available_destinations[guid]
+            if "@locked" in d:
+                dest_name = dest_name + " <LOCKED>"
+            destination_dict[guid] = dest_name
+        return destination_dict
+
+    def add_destination(self, destination_guid):
+        """Adds a destination to be used by this backup set. Raises an exception if
+        the supplied destination guid is not available to the parent device/org.
+
+        Args:
+            destination_guid (str, int): The globally unique identifier of the
+            destination to be added.
+        """
+        destination_guid = str(destination_guid)
+        if destination_guid in self._manager.available_destinations:
+            if destination_guid not in self.destinations:
+                self.data["destinations"].append({"@id": destination_guid})
+                self._changes["destinations"] = show_change(
+                    self._orig_destinations, self.destinations
+                )
+        else:
+            raise Py42Error(
+                "Invalid destination guid or destination not offered to device's Org."
+            )
+
+    def remove_destination(self, destination_guid):
+        """Removes a destination from use by this backup set.
+
+        Args:
+            destination_guid (str, int): The globally unique identifier of the
+            destination to be removed.
+        """
+        destination_guid = str(destination_guid)
+        self._raise_if_invalid_destination(destination_guid)
+        if destination_guid in self.destinations:
+            for d in self.data["destinations"]:
+                if d["@id"] == destination_guid:
+                    self.data["destinations"].remove(d)
+            self._changes["destinations"] = show_change(
+                self._orig_destinations, self.destinations
+            )
+
+    def lock_destination(self, destination_guid):
+        """Locks an in-use destination, disallowing the device owner from removing this
+        destination from their backup. Raises an exception if the supplied destination
+        guid is not in use on this backup set, or not available to the parent device/org.
+        """
+        destination_guid = str(destination_guid)
+        if destination_guid in self._manager.available_destinations:
+            if destination_guid not in self.destinations:
+                raise Py42Error("Destination is not added to device, unable to lock.")
```

Comment:
The exception is als re-used a bunch, which could make it complicated to maintain if we need to change some of it's text, so it should be re-usable somehow, either via custom exception or a local method to raise this error

Reply:
done